### ISSUE TITLE
Restoring optimizer state in load_checkpoint?

### ISSUE THREAD

**Issue by ipod825** [kevinzakka/recurrent-visual-attention#8]
In `load_checkpoint` [function](https://github.com/kevinzakka/recurrent-visual-attention/blob/f030a532cf85920e902ce3c22a8e70baaf20133f/trainer.py#L530), shouldn't you also load the optimizer state?

**Comment by kevinzakka** [comment]
@ipod825 I'm using plain SGD with momentum, and I'm tracking the learning rate which is getting reduced when the validation loss plateaus. I wasn't aware I needed to still save the optim state. Any reason why?

**Comment by ipod825** [comment]
Oh. Never mind. I wasn't so sure about how momentum works. You surely don't need to store other info. That said, it it is still better to store the full state or at least have a comment in case others want to play with your code and try another optimizer.

**Comment by kevinzakka** [comment]
@ipod825 Totally agree, I'll update the code and README to reflect that :) Thank you!

**Comment by kevinzakka** [comment]
@ipod825 so it turns out momentum update relies on the previous velocity value and I'd completely forgotten about it. So yes, you're absolutely right that I should save the optimizer dict state and I've updated the code accordingly. Thanks for pointing it out 😃
